File: project-nexus-portal/fe/src/services/academicService.js
import api from './api';

export const academicService = {
  getAllFaculties: async () => {
    const response = await api.get('/academic/faculties');
    return response.data;
  },

  getDepartmentsByFaculty: async (facultyId) => {
    if (!facultyId) return [];
    const response = await api.get(`/academic/departments?facultyId=${facultyId}`);
    return response.data;
  },

  getAllDepartments: async () => {
    const response = await api.get('/academic/departments/all');
    return response.data;
  },

  getMajorsByDepartment: async (departmentId) => {
    if (!departmentId) return [];
    const response = await api.get(`/academic/majors?departmentId=${departmentId}`);
    return response.data;
  },

  getAllMajors: async () => {
    const response = await api.get('/academic/majors/all');
    return response.data;
  },

  getAllCohorts: async () => {
    const response = await api.get('/academic/cohorts');
    return response.data;
  },
};

export default academicService;

File: project-nexus-portal/be/src/main/java/com/nexus/portal/service/AcademicService.java
package com.nexus.portal.service;

import com.nexus.portal.dto.response.CohortResponse;
import com.nexus.portal.dto.response.DepartmentResponse;
import com.nexus.portal.dto.response.FacultyResponse;
import com.nexus.portal.dto.response.MajorResponse;

import java.util.List;

public interface AcademicService {
    List<FacultyResponse> getAllFaculties();
    List<DepartmentResponse> getAllDepartments();
    List<DepartmentResponse> getDepartmentsByFaculty(Long facultyId);
    List<MajorResponse> getAllMajors();
    List<MajorResponse> getMajorsByDepartment(Long departmentId);
    List<CohortResponse> getAllCohorts();
}
